### PR TITLE
SF-2581 Add missing target books message when selecting drafting books

### PR DESCRIPTION
**Pull request by pmachapman**
This PR adds a message to the draft generation steps that lists books not found in the target text, but which are in the source text (and so the user would expect to be able to translate them).

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370)

### CODE REVIEW

**Review by nigel-wells — APPROVED** [review]
This looks good Peter - just one query.

> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-:-NsuV56r05R2euEGgiRE:b-p9xle5)* status: all files reviewed, 1 unresolved discussion (waiting on @pmachapman)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation-steps/draft-generation-steps.component.html` line 68 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-NsuUUj19pVw-4ew1ch5:-NsuUUj19pVw-4ew1ch6:bwdcz6x) ([raw file](https://github.com/sillsdev/web-xforge/blob/3077749b214f466d1f91f80325e50685b61ecf37/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation-steps/draft-generation-steps.component.html#L68)):*
> ```HTML
>         ></app-book-multi-select>
> 
>         <app-notice *ngIf="unusableTrainingSourceBooks.length || unusableTrainingTargetBooks.lastIndexOf" type="light">
> ```

Why did you use `lastIndexOf` here but `length` in the previous step as well as `length` in the `ngIf` below?

**Review by pmachapman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370)* status: all files reviewed, 1 unresolved discussion (waiting on @nigel-wells)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation-steps/draft-generation-steps.component.html` line 68 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-NsuUUj19pVw-4ew1ch5:-NsuZ69uC0Bx36iffRuk:b-fanpx9) ([raw file](https://github.com/sillsdev/web-xforge/blob/3077749b214f466d1f91f80325e50685b61ecf37/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-generation-steps/draft-generation-steps.component.html#L68)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Why did you use `lastIndexOf` here but `length` in the previous step as well as `length` in the `ngIf` below?
</blockquote></details>

Fixed. Good spotting - autocomplete must have tricked me!

**Comment by Nateowami** [comment]
I think we need to discuss before merging. I think there's been a misunderstanding of the purpose/need. Or maybe I'm mistaken. But let's hold off for a bit until we can discuss.

**Comment by pmachapman** [comment]
@Nateowami Based on the discussions in the MT drafting dev interlock meeting last week, are you OK with this PR being merged, or do you think it needs further work or clarification?

**Review by nigel-wells — APPROVED** [review]
It's better @josephmyers but I find myself looking at my screen with 4 notice panels and the one panel I probably care about the most is the hardest to spot:

Can we try some of the other colors to make it more noticeable? This isn't a blocker but it could be improved.

> Reviewed 14 of 14 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-:-NvjXlBX9qZYtL2AZpaY:b-47c3dx)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Nateowami)

**Review by Nateowami — COMMENTED** [review]
@nigel-wells I don't see an obvious way to improve this. Any time you make a notice more prominent, it seems you're starting an arms race for being more noticeable. The other ones will become less noticeable in comparison.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-:-NvmjEAN1xTS-mwVOZ1G:b-xdig77)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @pmachapman)

**Review by nigel-wells — APPROVED** [review]
Yes, which highlights (excuse the pun) the issue we have - too many notices are showing. 

Do we even need the notice about the project syncing - surely that could be included either just before you start the draft or as part of the draft progress.

Do we need the one talking about 200 supporting languages highlighted? Sure, if there is an error, but does it matter enough to make it a notice vs being part of the intro paragraph if at all?

We don't need to address it as part of this PR but we could do something as part of an overhaul of this UI IMO.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2370#-:-Nvo1mD0CKgspup74ov0:b-ooz1ab)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @pmachapman)

**Comment by josephmyers** [comment]
Thanks, Nigel. I absolutely agree -- we need to circle back to this during our UX meeting tomorrow. This workflow/experience needs some rework.

**Pull request merged**
